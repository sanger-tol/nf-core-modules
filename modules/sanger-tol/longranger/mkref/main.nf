process LONGRANGER_MKREF {
    tag "${meta.id}"
    label 'process_medium'

    // NOTE: As longranger is proprietary software, no container or conda environment
    // can be provided for this module. Please configure your own container and then
    // pass the path to this container as part of a Nextflow configuration file.

    input:
    tuple val(meta), path(assembly)

    output:
    tuple val(meta), path("refdata-*"), emit: reference
    path "versions.yml"               , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    if (workflow.profile.tokenize(',').intersect(['conda', 'mamba']).size() >= 1) {
        error "Error: longranger does not support Conda. Please use Docker / Singularity instead."
    }
    """
    longranger mkref ${assembly}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        longranger: \$(longranger mkref --version | grep longranger | sed 's/.*(//' | sed 's/).*//')
    END_VERSIONS
    """

    stub:
    def prefix = fasta - ~/\.fa/
    """
    mkdir refdata-${prefix}
    mkdir -p refdata-${prefix}/fasta
    mkdir -p refdata-${prefix}/genes
    mkdir -p refdata-${prefix}/regions
    mkdir -p refdata-${prefix}/snps

    touch refdata-${prefix}/fasta/genome.fa
    touch refdata-${prefix}/fasta/genome.fa.amb
    touch refdata-${prefix}/fasta/genome.fa.ann
    touch refdata-${prefix}/fasta/genome.fa.bwt
    touch refdata-${prefix}/fasta/genome.fa.fai
    touch refdata-${prefix}/fasta/genome.fa.flat
    touch refdata-${prefix}/fasta/genome.fa.gdx
    touch refdata-${prefix}/fasta/genome.fa.pac
    touch refdata-${prefix}/fasta/genome.fa.sa

    touch refdata-${prefix}/genome

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        longranger: \$(longranger mkref --version | grep longranger | sed 's/.*(//' | sed 's/).*//')
    END_VERSIONS
    """
}
