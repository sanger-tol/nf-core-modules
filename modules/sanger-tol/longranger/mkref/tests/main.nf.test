nextflow_process {

    name "Test Process LONGRANGER_MKREF"
    script "../main.nf"
    process "LONGRANGER_MKREF"
    config "./nextflow.config"

    tag "modules"
    tag "modules_sangertol"
    tag "longranger"
    tag "longranger/mkref"

    test("meles meles - fasta") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.modules_testdata_base_path + 'Meles_meles/assembly/release/mMelMel3.1_paternal_haplotype/GCA_922984935.2.subset.fasta.gz', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("meles meles - fasta - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.modules_testdata_base_path + 'Meles_meles/assembly/release/mMelMel3.1_paternal_haplotype/GCA_922984935.2.subset.fasta.gz', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
