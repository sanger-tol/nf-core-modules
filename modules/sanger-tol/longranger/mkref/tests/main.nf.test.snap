{
    "undibacterium unclassified": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            [
                                "genome.fa:md5,e5c01369a8bb02740acc39734d6abfcd",
                                "genome.fa.amb:md5,37f8bdd8b35c5dcae03bd83e9e2490e2",
                                "genome.fa.ann:md5,d686edce9d3d907f226fe73f895e161b",
                                "genome.fa.bwt:md5,0572f320ccc9fbc68edcffb3ef162b0e",
                                "genome.fa.fai:md5,feb391acd125f11289949368f10f7f79",
                                "genome.fa.flat:md5,2837f7616db364201c813d503ea9c230",
                                "genome.fa.gdx:md5,058439320db484c083c23d5dbf1c5c95",
                                "genome.fa.pac:md5,25ac263cd40c04a89ed1192187d1adbd",
                                "genome.fa.sa:md5,09a1677f1a75c3ea611fd80d8927b89b"
                            ],
                            [
                                
                            ],
                            "genome:md5,920cfb4f613de6917b8562878b7ed08b",
                            [
                                
                            ],
                            [
                                
                            ]
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,24bd8f4ee059041a932b4d70bbcfb43a"
                ],
                "reference": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            [
                                "genome.fa:md5,e5c01369a8bb02740acc39734d6abfcd",
                                "genome.fa.amb:md5,37f8bdd8b35c5dcae03bd83e9e2490e2",
                                "genome.fa.ann:md5,d686edce9d3d907f226fe73f895e161b",
                                "genome.fa.bwt:md5,0572f320ccc9fbc68edcffb3ef162b0e",
                                "genome.fa.fai:md5,feb391acd125f11289949368f10f7f79",
                                "genome.fa.flat:md5,2837f7616db364201c813d503ea9c230",
                                "genome.fa.gdx:md5,058439320db484c083c23d5dbf1c5c95",
                                "genome.fa.pac:md5,25ac263cd40c04a89ed1192187d1adbd",
                                "genome.fa.sa:md5,09a1677f1a75c3ea611fd80d8927b89b"
                            ],
                            [
                                
                            ],
                            "genome:md5,920cfb4f613de6917b8562878b7ed08b",
                            [
                                
                            ],
                            [
                                
                            ]
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,24bd8f4ee059041a932b4d70bbcfb43a"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.7"
        },
        "timestamp": "2025-10-03T16:15:43.575281665"
    },
    "undibacterium unclassified - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            [
                                "genome.fa:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "genome.fa.amb:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "genome.fa.ann:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "genome.fa.bwt:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "genome.fa.fai:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "genome.fa.flat:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "genome.fa.gdx:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "genome.fa.pac:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "genome.fa.sa:md5,d41d8cd98f00b204e9800998ecf8427e"
                            ],
                            [
                                
                            ],
                            "genome:md5,d41d8cd98f00b204e9800998ecf8427e",
                            [
                                
                            ],
                            [
                                
                            ]
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,24bd8f4ee059041a932b4d70bbcfb43a"
                ],
                "reference": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            [
                                "genome.fa:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "genome.fa.amb:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "genome.fa.ann:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "genome.fa.bwt:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "genome.fa.fai:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "genome.fa.flat:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "genome.fa.gdx:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "genome.fa.pac:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "genome.fa.sa:md5,d41d8cd98f00b204e9800998ecf8427e"
                            ],
                            [
                                
                            ],
                            "genome:md5,d41d8cd98f00b204e9800998ecf8427e",
                            [
                                
                            ],
                            [
                                
                            ]
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,24bd8f4ee059041a932b4d70bbcfb43a"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.7"
        },
        "timestamp": "2025-10-03T16:16:14.053862305"
    }
}